### ISSUE TITLE
Gitlab pipeline reports 404 error while editing pipeline configuration

### ISSUE THREAD

**Issue by happyyangyuan** [rancher/rancher#16950]
**Steps to reproduce :**
1、A private installed gitlab's application is setup successfully. 
2、Enable a gitlab repo.
3、Edit the pipeline configuration.

**Result:**
The pipeline configuration page displays '404 not found' error.
Rancher server log shows 
````
[ERROR] PipelineController p-kf5rz/p-jhrh5 [pipeline-controller] failed with :  404: {"error":"404 Not Found"}
````

**Other details that may be helpful:**
Is my gitlab version too old?

**Environment information**
- Rancher version:   v2.1.3
- Installation option : single
- Gitlab version: GitLab Community Edition 10.2.2 da70bc4

**Comment by gitlawr** [comment]
I think the gitlab version is irrelevant for the error, given that you've configured the auth and fetched/enabled the repos.

1. Are these steps done using the same gitlab user credential, i.e. operated by the same rancher user?
2. Is the gitlab host url correctly set, i.e. you can clone from the repo link shown in `Edit Pipeline Configuration` page? 

It would also be nice if you can provide info/structure of the repo url. There is an issue related to gitlab subgroup/project so we can see if they are relevant.

**Comment by kontra** [comment]
Get same error.
Rancher: 2.2.2
GitLab: 11.10.4
Gitlab server url: http://gitlab-server.local/gitlab/
After configuring repository, rancher shows projects, allows to enable.
But after clicking Edit Config on pipline throws {"message":"404 Project Not Found"}
I mirrored Gitlab on another server to the root url, Rancher works with it fine.
So I think the problem is maybe from "/gitlab/" suffix in url.

**Comment by happyyangyuan** [comment]
I agree with you. My former gitlab's base url is `https://dev.xyz123456.com/git/` with the prefix `/git/` and the same error.

**Comment by Kuchenm0nster** [comment]
Any progress? 
I'm facing the same problem.
with domain.tld/gitlab I can find all Projects and enable them but if I try to edit them no branches are found just 404
